Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/exception/InvalidTestCertificateException.kt

```diff
@@ -0,0 +1,90 @@
+package de.rki.coronawarnapp.covidcertificate.exception
+
+import android.content.Context
+import de.rki.coronawarnapp.util.HumanReadableError
+import de.rki.coronawarnapp.util.ui.CachedString
+import de.rki.coronawarnapp.util.ui.LazyString
+
+class InvalidTestCertificateException(errorCode: ErrorCode) : InvalidHealthCertificateException(errorCode) {
+    override fun toHumanReadableError(context: Context): HumanReadableError {
+        var errorCodeString = errorCode.toString()
+        errorCodeString = if (errorCodeString.startsWith(PREFIX_TC)) errorCodeString else PREFIX_TC + errorCodeString
+        return HumanReadableError(
+            description = errorMessage.get(context) + "\n\n$errorCodeString"
+        )
+    }
+
+    override val errorMessage: LazyString
+        get() = when (errorCode) {
+
+            ErrorCode.DCC_COMP_NO_NETWORK,
+            ErrorCode.PKR_NO_NETWORK -> CachedString { context ->
+                context.getString(ERROR_MESSAGE_NO_NETWORK)
+            }
+
+            ErrorCode.DCC_COMP_202 -> CachedString { context ->
+                context.getString(ERROR_MESSAGE_TRY_AGAIN_DCC_NOT_AVAILABLE_YET)
+            }
+
+            ErrorCode.DCC_COMP_410 -> CachedString { context ->
+                context.getString(ERROR_MESSAGE_DCC_EXPIRED)
+            }
+
+            // TODO
+/*            ErrorCode.HC_BASE45_DECODING_FAILED,
+            ErrorCode.HC_CBOR_DECODING_FAILED,
+            ErrorCode.HC_COSE_MESSAGE_INVALID,
+            ErrorCode.HC_ZLIB_DECOMPRESSION_FAILED,
+            ErrorCode.HC_COSE_TAG_INVALID,
+            ErrorCode.HC_CWT_NO_DGC,
+            ErrorCode.HC_CWT_NO_EXP,
+            ErrorCode.HC_CWT_NO_HCERT,
+            ErrorCode.HC_CWT_NO_ISS,
+            ErrorCode.JSON_SCHEMA_INVALID,*/
```

Comment:
is it needed later or can be deleted ?

Reply:
I will go through all error handling again with ticket 7507, this is just a reminder for me to check those codes